### PR TITLE
MDEV-20685/MDEV-21595: solaris compile fixes

### PR DESCRIPTION
**Pull request by grooverdan**
This cleans up to get a compilable and running tests with:

`cmake  -DCONC_WITH_{UNITTEST,SSL}=OFF -DWITH_UNIT_TESTS=OFF -DCMAKE_BUILD_TYPE=Debug -DPLUGIN_{ARCHIVE,TOKUDB,MROONGA,OQGRAPH,ROCKSDB,CONNECT,SPIDER,SPHINX,AUTH_PAM}=NO ../mariadb-server-10.2`

### CODE REVIEW

**Comment by grooverdan** [comment]
If the MDEV-21595 is going to be too much to merge, maybe this hack https://github.com/OpenIndiana/oi-userland/blob/oi/hipster/components/database/mariadb-103/patches/11-offset_t.patch and merge a renaming of variables in 10.5

**Comment by grooverdan** [comment]
note: [sigreturn should apply to OSX too](https://travis-ci.org/github/grooverdan/mariadb-server/jobs/671173637)

File: storage/innobase/include/rem0types.h [hunk 1]
```diff
@@ -31,7 +31,7 @@ Created 5/30/1994 Heikki Tuuri
 typedef byte	rec_t;
 
 /** This type represents a field offset in a rec_t* */
-typedef unsigned short int offset_t;
+typedef unsigned short int rem0offset_t;
```
Comment (robertbindar) [inline storage/innobase/include/rem0types.h:34]: Given this name appear in a loooot of places, I believe the name should be as intuitive as possible. Marko suggested `rec_offs`, sounds good enough for me. @kevgs has the final word on this as he did the refactoring.

**Comment by dr-m** [comment]
@kevg @grooverdan I made a suggestion in [MDEV-21595](https://jira.mariadb.org/browse/MDEV-21595):
```sh
perl -i -pe 's/\<offset_t\>/rec_offs/g' $(git grep -w offset_t storage/innobase)
```
Idea: with equal-length type name, no code will need to be reformatted or reindented.

**Comment by kevgs** [comment]
In C++ such name collisions should be resolved with namespaces. But introducing a namespace will require a lot of code changes which we can't do automatically. So I'd stick with what Marko suggested. Current name with zero in the middle is a no-go, IMO.

**Comment by grooverdan** [comment]
connector-c fixes https://github.com/mariadb-corporation/mariadb-connector-c/pull/134 and https://github.com/mariadb-corporation/mariadb-connector-c/pull/133 are merged. one https://github.com/mariadb-corporation/mariadb-connector-c/pull/132 gssapi outstanding.

**Comment by grooverdan** [comment]
@dr-m, @kevgs , @robertbindar thanks for the review.

Thanks for the tip about the type replacement. Corrected perl command in commit message in case of upbranch merge conflicts.

**Comment by robertbindar** [comment]
Hi @grooverdan, tests seem legit, I'm ready to merge this.
What do you say about squashing the commits so that we have 1 for MDEV-21595 and 1 for MDEV-20685. Whilst for the two of us, it makes total sense to have this clear division, for other developers it would be more digestible to see only one or two commits suggesting they fix all compiling errors on solaris.

**Comment by robertbindar** [comment]
P.S. Excellent that you already have most compile errors fixed in the C connector, many thanks for that as well.

**Comment by grooverdan** [comment]
Thanks @robertbindar

**Pull request merged**